vcf/sample_filter.py

```diff
@@ -81,13 +81,13 @@ def filt2idx(item):
                 # is int, check if it's an idx
                 if item < len(self.samples):
                     return item
-        filters = set(filter(lambda x: x is not None, map(filt2idx, filt_s)))
+        filters = set([x for x in map(filt2idx, filt_s) if x is not None])
```

Comment:
```suggestion
        filters = set(x for x in map(filt2idx, filt_s) if x is not None)
```
or if you don't mind actual set notation:
```suggestion
        filters = {x for x in map(filt2idx, filt_s) if x is not None}
```